components/project-list/project-list.jsx

```diff
@@ -16,28 +16,48 @@ export default React.createClass({
       displayBatchIndex: 1
     };
   },
+  getDefaultProps() {
+    return {
+      params: {
+        featured: null,
+        issue: null,
+        search: null,
+        ids: null
+      }
+    };
+  },
   componentDidMount() {
     this.fetchData(this.props.params);
   },
   componentWillReceiveProps(nextProps) {
+    // Reset state
+    this.resetState();
```

Comment:
We should reset state as soon as a new search has been initiated. However, instead of showing "0 results found" I've update that message to `Searching for entries that match ‘some_keyword’...` (see `render()` below)